app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -68,6 +69,34 @@ def _validate_feature_value_against_schema(self, schema, value):
 
     def validate(self, data):
         data = super().validate(data)
+        data = self._validate_duplicate_branch_names(data)
+        data = self._validate_feature_configs(data)
+        return data
+
+    def _validate_duplicate_branch_names(self, data):
+        if "reference_branch" in data and "treatment_branches" in data:
+            ref_branch_name = data["reference_branch"]["name"]
+            treatment_branch_names = [
+                branch["name"] for branch in data["treatment_branches"]
+            ]
+            all_names = [ref_branch_name, *treatment_branch_names]
+            unique_names = set(all_names)
+
+            if len(all_names) != len(unique_names):
+                raise serializers.ValidationError(
+                    {
+                        "reference_branch": {
+                            "name": NimbusConstants.ERROR_DUPLICATE_BRANCH_NAME
```

Comment:
Hmm, might be better if this were a general global error rather than throwing this on all branches and not just the duplicate branches. But, I can't remember off the top of my head if that's actually possible within this serializer.

Reply:
Hrmm... Could go either way for me on the preference there. I'll leave it for now, we could explore in a follow up if we'd like.